src/QirRuntime/lib/QIR/intrinsicsOut.cpp

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <iostream>
+
+#include "qirTypes.hpp"
+#include "quantum__qis.hpp"
+
+extern "C"
+{
+    void quantum__qis__message__body(QirString* qstr)   // NOLINT
+    {
+        std::cout << qstr->str << std::endl;
```

Comment:
Dumping to cout is OK short term, but going forward it might not be sufficient and should probably be configurable through QirExecutionContext. Check with Cesar re requirements for output in Azure environment (IQ# also might have special needs here).

Reply:
Tagging in @anjbur for IQ# discussion.

Reply:
I have updated my branch and addressed all the concerns except the last Chris' email about `Extensions.cs#L96`. I'll take a look at the email in more details tomorrow.

Reply:
To unblock Guen, I decided to postpone the changes (if any) inspired by Chris' email. 
I filed #518 to keep track of that. Marking this discussion as resolved in this PR.